src/Wallet.purs

```diff
@@ -35,121 +36,46 @@ import Helpers ((<<>>))
 import Serialization as Serialization
 import Serialization.Address (Address, addressFromBytes)
 import Types.ByteArray (byteArrayToHex)
-import Types.CborBytes
-  ( CborBytes
-  , cborBytesToHex
-  , rawBytesAsCborBytes
-  )
+import Types.CborBytes (CborBytes, cborBytesToHex, rawBytesAsCborBytes)
 import Types.RawBytes (RawBytes, hexToRawBytes)
-import Types.Transaction (TransactionHash(TransactionHash))
 import Untagged.Union (asOneOf)
 
--- At the moment, we only support Nami's wallet. In the future we will expand
--- this with more constructors to represent out-of-browser wallets (e.g. WBE)
-data Wallet = Nami NamiWallet
-
--------------------------------------------------------------------------------
--- Nami backend
--------------------------------------------------------------------------------
--- Record-of-functions for real or mocked Nami wallet, includes `Ref` to
--- connection (e.g. with `window.cardano.nami` as a `NamiConnection`)
-type NamiWallet =
-  { -- A reference to a connection with Nami, i.e. `window.cardano.nami`
-    connection :: Ref.Ref NamiConnection
+type Cip30Wallet =
+  { -- A reference to a connection with the wallet, i.e. `window.cardano.nami`
+    connection :: Ref.Ref Cip30Connection
```

Comment:
Since this PR totally reworks the `Wallet` type, we can actually remove the `Ref` and hold the connection directly. The original idea was to have a ref to the API object since it will mutate if you call `enable` on it. But we actually call `enable` directly during initialization and then use the resulting API object, so we never need to `write` the ref. We can just totally eliminate the overhead of having a `Ref` hold it

Apologies for not pointing this out earlier